docs/contributing/03-stories/story-16.md

```diff
@@ -0,0 +1,100 @@
+# User Story 16 - Wing for TypeScript
+
+> **Status**: Draft
+
+Anders loves TypeScript, but he also loves the amazing experience Wing offers for building cloud applications.
+
+It's Andres' lucky day because Wing Cloud had just announced that very soon TypeScript developers are able to start leveraging Wing libraries from the comfort of
+their favorite programming language.
+
+He got an invite for a pre-release. The invite explains that initially, there is only support for `cloud.Bucket` and `cloud.Function` and for consuming [winglibs](https://github.com/winglang/winglibs).
+
+The example looks like this:
+
+First, install some libraries:
+
+```sh
+$ npm i winglang wingsdk @winglibs/cloud @winglibs/checks
+```
+
+```ts
+// main.ts
+import { inflight, App } from "wingsdk";
```

Comment:
Is this intentionally not `@winglang/sdk`?

```suggestion
import { inflight, App } from "@winglang/sdk";
```

I can see the benefit of a new package specifically for TS, but not sure it's an initial requirement.

Reply:
Yes, it is intentional - this is *not* the internal `@winglang/sdk` library which we have today. It's a new library designed to support the TypeScript experience.

I am proposing to separate these layers so that we don't couple our TypeScript experience with the JSII library we use as the underlying implementation for winglang's standard library.

They are not the same thing in my mind, and I hope that at some point we will completely deprecate `@winglang/sdk` and rewrite it winglang.

Reply:
Changed the name to a temporary name `wing4ts` so that it's clearer that this is not the same as `@winglang/sdk`.

Reply:
> I am proposing to separate these layers so that we don't couple our TypeScript experience with the JSII library we use as the underlying implementation for winglang's standard library.

Being in the same library doesn't necessarily couple them, although I would argue `App` maybe should be coupled since it will already do the exact same thing we already have to deal with in wing compiled output.
From a maintenance perspective, it's very useful if the codepath for the compiled ts and the jsified wing are similar. **As long as it doesn't harm or dictate the wing experience**, of course.

`inflight` is special, sure, but an entire library just for that seems a bit overkill. What if we just have a specific barrel import until the day comes where `@wingland/sdk` is deleted (which I'm assuming will take a while)

```
import { inflight, App } from "@winglang/sdk/typescript"; // or node or js or whatever
```

Where App can just be a re-export of the existing App.

---

In the end it's not a big deal, I just figured a new package can wait, but if you feel strongly about it being separate then I'm cool with that.

Reply:
Awesome. I think this separation will give us more flexibility to evolve the TypeScript experience independently of how the winglang standard library is implemented. I prefer TypeScript users not to be exposed to all the horrors within the internal implementation, and only have a nice and clean API to work with that's optimized for them.